wallet/wallet.go

```diff
@@ -230,6 +237,104 @@ func (w *SingleAddressWallet) SignTransaction(cs consensus.State, txn *types.Tra
 	return nil
 }
 
+// Redistribute returns a transaction that redistributes money in the wallet by
+// selecting a minimal set of inputs to cover the creation of the requested
+// outputs. It also returns a list of output IDs that need to be signed.
+//
+// NOTE: we can not reuse 'FundTransaction' because it randomizes the unspent
+// transaction outputs it uses and we need a minimal set of inputs
+func (w *SingleAddressWallet) Redistribute(cs consensus.State, outputs int, amount, feePerByte types.Currency, pool []types.Transaction) (types.Transaction, []types.OutputID, error) {
+	// prepare all outputs
+	var txn types.Transaction
```

Comment:
this whole function should be protected by a lock, like `FundTransaction` is

Reply:
oof embarrassing - thanks for catching that